apps/api/src/domain/models/allowedContexts/getAllowedContextsForModel.spec.ts

```diff
@@ -0,0 +1,49 @@
+import { resourceTypes } from '../../types/resourceTypes';
+import { EdgeConnectionContextType } from '../context/types/EdgeConnectionContextType';
+import { getAllowedContextsForModel } from './isContextAllowedForGivenResourceType';
+
+describe('getAllowedContextsForModel', () => {
+    // Loop over every `ResourceType`
+    Object.values(resourceTypes).forEach((resourceType) =>
+        describe(`resource type: ${resourceType}`, () => {
+            const allowedContexts = getAllowedContextsForModel(resourceType);
+
+            it('should have a specified list of allowed context types', () => {
+                const numberOfResults = allowedContexts.length;
+
+                expect(numberOfResults).toBeGreaterThan(0);
+            });
+
+            it('should allow the general context', () => {
+                const isGeneralContextAllowed = allowedContexts.includes(
+                    EdgeConnectionContextType.general
+                );
+
+                expect(isGeneralContextAllowed).toBe(true);
+            });
+
+            describe('the list of allowed contexts', () => {
+                it('should contain no duplicates', () => {
+                    const lengthOfList = allowedContexts.length;
+
+                    const numberOfUniqueElements = [...new Set(allowedContexts)].length;
+
+                    expect(numberOfUniqueElements).toBe(lengthOfList);
+                });
+            });
+        })
+    );
+
+    // Loop over every `EdgeConnectionContextType`
+    Object.values(EdgeConnectionContextType).forEach((contextType) =>
+        describe(`Edge connection context type: ${contextType}`, () => {
+            it(`is registered as the allowed context for at least one resource type`, () => {
+                const isContextTypeAllowedForSomeResourceType = Object.values(resourceTypes).some(
```

Comment:
I always have to remind myself what `.some()` does.  Maybe it should be called `.oneOrMore()`.  Have to put a rename request in to MDN or something...

Reply:
Because a breaking change that API won't take down the web just to make Geoff happy...